django/contrib/admin/static/admin/js/SelectBox.js

```diff
@@ -125,11 +121,8 @@
             } );
         },
         select_all: function(id) {
-            var box = document.getElementById(id);
-            for (var i = 0; i < box.options.length; i++) {
-                box.options[i].selected = 'selected';
-            }
+            $('#' + id + ' option').prop('selected', 'selected');
```

Comment:
Simply caching `box.options` and `box.options.length` in local variables should be enough here.